lib/mix/tasks/swagger.generate.ex

```diff
@@ -26,44 +27,46 @@ defmodule Mix.Tasks.Phx.Swagger.Generate do
   defp top_level_namespace, do: Mix.Project.get().application()[:mod] |> elem(0) |> Module.split |> Enum.drop(-1) |> Module.concat
   defp app_name, do: Mix.Project.get().project()[:app]
   defp default_swagger_file_path, do: app_path() <> "swagger.json"
-  defp default_router_module, do: Module.concat([top_level_namespace(), :Web, :Router])
-
+  defp default_router_module_name, do: Module.concat([top_level_namespace(), :Web, :Router])
+  defp default_endpoint_module_name, do: Module.concat([top_level_namespace(), :Web, :Endpoint])
+  defp router_module(switches), do: switches |> Keyword.get(:router, default_endpoint_module_name()) |> attempt_load
+  defp endpoint_module(switches), do: switches |> Keyword.get(:endpoint, default_endpoint_module_name()) |> attempt_load
+  
   def run(args) do
     Mix.Task.run("compile")
     Mix.Task.reenable("phx.swagger.generate")
     Code.append_path("#{app_path()}_build/#{Mix.env}/lib/#{app_name()}/ebin")
     {switches, params, _unknown} = OptionParser.parse(
       args,
-      switches: [router: :string, help: :boolean],
-      aliases: [r: :router, h: :help])
-
+      switches: [router: :string, endpoint: :string, help: :boolean],
+      aliases: [r: :router, e: :endpoint, h: :help])
+    
+    router = router_module(switches)
+    endpoint = endpoint_module(switches)
     cond do
       (Keyword.get(switches, :help)) ->
         usage()
-      has_no_endpoint() ->
-        IO.puts "Skipping app #{app_name()}, no Endpoint configured."
+      is_nil(router) ->
+        Logger.error "Skipping app #{app_name()}, no Router configured."
```

Comment:
`Logger.error` is probably too severe here. If `phx.swagger.generate` is run from the root of an umbrella application, then we expect some applications to be skipped.

`Logger.warn` should be fine.